app/src/main/java/io/appium/uiautomator2/utils/AlertHelpers.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.uiautomator2.utils;
+
+import android.support.annotation.Nullable;
+import android.support.test.uiautomator.By;
+import android.support.test.uiautomator.UiObject2;
+import android.widget.Button;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import io.appium.uiautomator2.common.exceptions.InvalidElementStateException;
+import io.appium.uiautomator2.common.exceptions.NoAlertOpenException;
+
+import static android.text.TextUtils.join;
+import static io.appium.uiautomator2.utils.Device.getUiDevice;
+import static org.apache.commons.lang.StringUtils.isBlank;
+
+public class AlertHelpers {
+    private static final String alertButtonResIdPattern = "^android:id/button\\d+$";
+    private static final String alertContentResId = "android:id/content";
```

Comment:
So should we consider any view with id ```context``` as an alert?
![image](https://user-images.githubusercontent.com/3530074/39543518-1b170c80-4e54-11e8-944b-484caa2f9ab2.png)

Reply:
yep, changed it